File: src/types/location.ts
export interface Location {
  id: string;
  project_id: string;
  name: string;
  location_type: string | null;
  description: string | null;
  appearance: string | null;
  atmosphere: string | null;
  inhabitants: string | null;
  notes: string | null;
  map_path: string | null;
  tags: string | null;
  created_at: string;
  updated_at: string;
}

export interface CreateLocationInput {
  project_id: string;
  name: string;
  location_type?: string | null;
  description?: string | null;
  appearance?: string | null;
  atmosphere?: string | null;
  inhabitants?: string | null;
  notes?: string | null;
  map_path?: string | null;
  tags?: string | null;
}

export interface UpdateLocationInput {
  name?: string;
  location_type?: string | null;
  description?: string | null;
  appearance?: string | null;
  atmosphere?: string | null;
  inhabitants?: string | null;
  notes?: string | null;
  map_path?: string | null;
  tags?: string | null;
}

File: src-tauri/src/models/location.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Location {
    pub id: String,
    pub project_id: String,
    pub name: String,
    pub location_type: Option<String>,
    pub description: Option<String>,
    pub appearance: Option<String>,
    pub atmosphere: Option<String>,
    pub inhabitants: Option<String>,
    pub notes: Option<String>,
    pub map_path: Option<String>,
    pub tags: Option<String>,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CreateLocationInput {
    pub project_id: String,
    pub name: String,
    pub location_type: Option<String>,
    pub description: Option<String>,
    pub appearance: Option<String>,
    pub atmosphere: Option<String>,
    pub inhabitants: Option<String>,
    pub notes: Option<String>,
    pub map_path: Option<String>,
    pub tags: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct UpdateLocationInput {
    pub name: Option<String>,
    pub location_type: Option<String>,
    pub description: Option<String>,
    pub appearance: Option<String>,
    pub atmosphere: Option<String>,
    pub inhabitants: Option<String>,
    pub notes: Option<String>,
    pub map_path: Option<String>,
    pub tags: Option<String>,
}
